File: PlateformRH/src/main/java/com/PlateformRH/Utilisateur/UtilisateurService.java
package com.PlateformRH.Utilisateur;

import java.util.List;
import java.util.Map;

public interface UtilisateurService {

    void create (utilisateur utilisateur);
    void delete (Long id);
    List<UtilisateurDTO> getAll();

    // update utilisateur
    UtilisateurDTO update(Long id, UtilisateurDTO utilisateur);

     void prepareUtilisateur(utilisateur utilisateur);
    List<UtilisateurDTO> getUsersByRole(String role);
    void changerMotDePasse(
            Long id,
            Map<String, String> body
    );
}

File: rh/src/app/services/utilisateur.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Utilisateur } from 'src/app/models/utilisateur';

@Injectable({
  providedIn: 'root'
})
export class UtilisateurService {

  private apiUrl = 'http://localhost:8000/utilisateurs';

  constructor(private http: HttpClient) {}

  // ================= GET ALL =================
  getAll(): Observable<Utilisateur[]> {
    return this.http.get<Utilisateur[]>(this.apiUrl);
  }

  // ================= GET BY ID =================
  getById(id: number): Observable<Utilisateur> {
    return this.http.get<Utilisateur>(`${this.apiUrl}/${id}`);
  }

  // ================= CREATE =================
  create(user: Utilisateur): Observable<void> {
    return this.http.post<void>(this.apiUrl, user);
  }

  // ================= UPDATE =================
  update(id: number, user: Utilisateur): Observable<Utilisateur> {
    return this.http.put<Utilisateur>(`${this.apiUrl}/${id}`, user);
  }

  // ================= DELETE =================
  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  // ================= GET BY ROLE =================
  getByRole(role: string): Observable<Utilisateur[]> {
    return this.http.get<Utilisateur[]>(`${this.apiUrl}/role/${role}`);
  }

}